pages/villagers/[id].jsx

```diff
@@ -0,0 +1,128 @@
+import { useRouter } from 'next/router';
+import StyledDetailedCard from '../../components-styled/StyledDetailedCard';
+import StyledImage from '../../components-styled/StyledImage';
+import StyledDetailFlexWrapper from '../../components-styled/StyledDetailFlexWrapper';
+import StyledCardDetails from '../../components-styled/StyledCardDetails';
+import StyledButton from '../../components-styled/StyledButton';
+import StyledSpan from '../../components-styled/StyledSpan';
+import StyledIconButtonWrapper from '../../components-styled/StyledIconButtonwrapper';
+import StyledIconButton from '../../components-styled/StyledIconButton';
+import StyledSvgWrapper from '../../components-styled/StyledSvgWrapper';
+import Favorite from '../../public/favorite.svg';
+import Neighbor from '../../public/neighbor.svg';
+import { useSelector, useDispatch } from 'react-redux';
+import Link from 'next/link';
+import {
+  selectVillagers,
+  toggleFavorites,
+  toggleNeighbors,
+} from '../../features/villagers/villagerSlice';
+
+import { useEffect } from 'react';
+import PageWrapper from '../../components/PageWrapper';
+import StyledHeader from '../../components-styled/StyledHeader';
+import StyledCardsWrapper from '../../components-styled/StyledCardsWrapper';
+
+export default function VillagerDetailPage() {
+  const router = useRouter();
+  const { id } = router.query;
+
```

Comment:
```suggestion
  const { id: idString } = router.query;
  const id = parseInt(idString)
```
und dann unten überall das parseInt raus